[SPARK-12941][SQL] Spark-SQL JDBC Oracle dialect fails to map string datatypes to Oracle VARCHAR datatype

thomastechs opened the pull request:
Adding a getJDBCType() method to the JdbcDialects.scala for OracleDialect,which would create a VARCHAR type for Oracle create table operations.Currently the type TEXT is throwing exceptions which is incompatible to Oracle

thomastechs committed:
Update for [SPARK-12941][SQL]

Adding a getJDBCType() method to the JdbcDialects.scala which would create a VARCHAR type for Oracle create table operations.Currently the type TEXT is throwing exceptions which is incompatible to Oracle

thomastechs commented:
@rxin Could you please review this fix?

yhuai commented:
test this please

yhuai commented:
@thomastechs Can you also create a pr for master? Usually we merge the one for master and backport it to older branches. If there is any conflict, we resolve that use an individual pr.

yhuai commented:
Is it possible to create a  OracleIntegrationSuite (like MySQLIntegrationSuite)?

yhuai commented:
just realized that it is against your own repo. Can you create a open source pr?

thomastechs commented:
@yhuai   The code change for 1.4 branch is in JdbcDialects.scala (as shown in this PR), where as the code change required for master would be in OracleDialect.scala, where the OracleDialect.scala is a separate file . So, would this PR be easier to merge? What do you think? I shall create an opensource pr

yhuai commented:
ok i see. Then, let's have individual prs. Can you submit prs to open source repo?

thomastechs commented:
@yhuai
I have created a new PR for 1.4 
https://github.com/apache/spark/pull/10912

thomastechs committed:
Update JdbcDialects.scala

Updated the style error-remove whitespace

thomastechs committed:
Update for [SPARK-12941][SQL] JDBCSuite.scala

Adding test case for the getJDBCType() method for the datatype-StringType as a part of OracleDialect

thomastechs committed:
Update JDBCSuite.scala [SPARK-12941][Branch-1.4]

Updated for the style fix.

thomastechs committed:
Update JDBCSuite.scala
